src/instruction/functions.rs

```diff
@@ -181,31 +181,41 @@ pub fn not(vm: &mut Vm, reg: u8) -> Result<()> {
     Ok(())
 }
 
-pub fn eq(vm: &mut Vm, left: u8, right: u8) -> Result<()> {
-    vm.prev_comp = vm.registers[left as usize]
-        .clone()
-        .is_equal(vm.registers[right as usize].clone(), &vm.gc)?;
-    vm.index += Index(1);
-
-    Ok(())
-}
-
-pub fn not_eq(vm: &mut Vm, left: u8, right: u8) -> Result<()> {
-    vm.prev_comp = !vm.registers[left as usize]
-        .clone()
-        .is_equal(vm.registers[right as usize].clone(), &vm.gc)?;
-    vm.index += Index(1);
-
-    Ok(())
-}
-
-pub fn greater_than(_vm: &mut Vm, _left: u8, _right: u8) -> Result<()> {
-    todo!()
+macro_rules! comparison_operator {
+    ( $op_name:ident, $compare:ident => $found_bool:expr, || $else_bool:expr ) => {
```

Comment:
Really a nitpicky thing, but I kinda think it'd look better with 
```rust
$op_name:ident, $compare:ident => $found_bool:expr || $else_bool:expr
```
(Note the missing comma after `found_bool`) because the resulting macro usage could be
```rust
comparison_operator!(eq, Equal => true || false);
```
Additionally, with
```rust
( ($ $op_name:ident, $compare:ident => $found_bool:expr, || $else_bool:expr ),* )
```
And wrapping the body in `$( ... )*` all macro usages could be done from one invocation, like so
```rust
comparison_operator! {
    eq,           Equal   => true  || false,
    not_eq,       Equal   => false || true,
    greater_than, Greater => true  || false,
    less_than,    Less    => true  || false
}
```